lib/openssl/mac.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+module OpenSSL
+  if defined?(MAC)
+    class MAC
+      def ==(other)
+        return false unless self.class === other
+        return false unless self.mac.bytesize == other.mac.bytesize
+
+        OpenSSL.fixed_length_secure_compare(self.mac, other.mac)
+      end
```

Comment:
Comparing MACs generated by different algorithms or parameters would be pointless, but this doesn't prevent it and it's probably not possible unless we store parameters or sensitive keys ourselves, which would be a bad idea.

Honestly, I'm not sure if this should be provided by this library.

Reply:
I've forgotten it was in `OpenSSL::HMAC`. It seems to have the same issue.

IMO we should not add it to `OpenSSL::MAC`, though we can't remove the existing method from `OpenSSL::HMAC`.